core/src/test/java/de/jplag/highlightextraction/frequencysimilarity/FrequencySimilarityTest.java

```diff
@@ -0,0 +1,186 @@
+
+    private List<TokenType> extractMatchTokenTypes(Match match) {
+        List<TokenType> tokens = new ArrayList<>();
+        for (int i = match.startOfFirst(); i <= match.startOfFirst() + match.lengthOfFirst(); i++) {
+            tokens.add(testSubmission.getTokenList().get(i).getType());
+        }
+        return tokens;
+    }
+
+    /**
+     * Tests frequency similarity strategies
+     */
+    @Test
+    @DisplayName("frequencySimilarity berechnet realistische Werte")
+    void testFrequencySimilarity_withRealisticMatchData() {
+
+        double resultProportional = similarityProportional.frequencySimilarity(comparison, 0.5);
+        double resultLinear = similarityLinear.frequencySimilarity(comparison, 0.5);
+        double resultQuadratic = similarityQuadratic.frequencySimilarity(comparison, 0.5);
+        double resultSigmoid = similaritySigmoid.frequencySimilarity(comparison, 0.5);
+
+        double roundedResultProportional = Math.round(resultProportional * 100.0) / 100.0;
+        double roundedResultLinear = Math.round(resultLinear * 100.0) / 100.0;
+        double roundedResultQuadratic = Math.round(resultQuadratic * 100.0) / 100.0;
+        double roundedResultSigmoid = Math.round(resultSigmoid * 100.0) / 100.0;
+
+        // assertEquals(1.0, roundedResultProportional, 0.01, "proportional frequencySimilarity should have following weight at
+        // 0.5: 1");
+        // assertEquals(1.0, roundedResultLinear, 0.01, "linear frequencySimilarity should have following weight at 0.5: 1");
+        // assertEquals(1.0, roundedResultQuadratic, 0.01, "quadratic frequencySimilarity should have following weight at 0.5:
+        // 1");
+        // assertEquals(1.0, roundedResultSigmoid, 0.01, "sigmoid frequencySimilarity should have following weight at 0.5: 1");
+        // List<JPlagComparison> comparisons = new ArrayList<>();
+        // comparisons.add(comparison);
+        // comparisons.add(comparison2);
+        // List<JPlagComparison> result5 = similarityProportional.calculateFrequencySimilarity(comparisons, 0);
+        // List<JPlagComparison> result6 = similarityLinear.calculateFrequencySimilarity(comparisons, 1);
+        // List<JPlagComparison> result7 = similarityQuadratic.calculateFrequencySimilarity(comparisons, 0.5);
+        // List<JPlagComparison> result8 = similaritySigmoid.calculateFrequencySimilarity(comparisons, 0.5);
+        // assertEquals(0.0, result5.get(1).similarity(), 0.01, "proportional frequencySimilarity should have following weight
+        // at 0.5: 1");
+        // assertEquals(0.0, result6.get(1).similarity(), 0.01, "linear frequencySimilarity should have following weight at 0.5:
+        // 1");
+        // assertEquals(0.0, result7.get(1).similarity(), 0.01, "quadratic frequencySimilarity should have following weight at
+        // 0.5: 1");
+        // assertEquals(0.0, result8.get(1).similarity(), 0.01, "sigmoid frequencySimilarity should have following weight at
+        // 0.5: 1");
+
```

Comment:
Remove out-commented code, please. There's no reason to commit something like this.